rmf_task/src/rmf_task/BackupFileManager.cpp

```diff
@@ -59,7 +62,17 @@ class BackupFileManager::Group::Implementation
   : group_directory(std::move(directory)),
     settings(std::move(settings))
   {
-    // Do nothing
+    std::filesystem::create_directories(this->group_directory);
+
+    // BH: Should we handle permission exceptions to give the user more info?
+    std::filesystem::permissions(this->group_directory,
+      std::filesystem::perms::owner_all | std::filesystem::perms::group_all,
+      std::filesystem::perm_options::add);
+
+   for(auto& p: std::filesystem::recursive_directory_iterator(this->group_directory))
+    std::filesystem::permissions(p,
```

Comment:
Style nitpick: wrap curly braces `{ }` around multi-line code blocks for `if`, `for`, etc.

Reply:
ff3227c0b3950d9bc5ad1199566afb92fe857d74 should resolve this